lib/compat/wordpress-6.2/class-gutenberg-rest-templates-controller-6-2.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Templates_Controller_6_2 class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Templates REST API Controller.
+ */
+class Gutenberg_REST_Templates_Controller_6_2 extends Gutenberg_REST_Templates_Controller {
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/lookup',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_template_fallback' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'slug'            => array(
+							'description' => __( 'The slug of the template to get the fallback for', 'gutenberg' ),
+							'type'        => 'string',
+							'required'    => true,
+						),
+						'is_custom'       => array(
+							'description' => __( 'Indicates if a template is custom or part of the template hierarchy', 'gutenberg' ),
+							'type'        => 'boolean',
+						),
+						'template_prefix' => array(
+							'description' => __( 'The template prefix for the created template. This is used to extract the main template type ex. in `taxonomy-books` we extract the `taxonomy`', 'gutenberg' ),
+							'type'        => 'string',
+						),
+						'ignore_empty'    => array(
```

Comment:
I wonder if we really need this flag or if we should do this by default?

Reply:
I added the flag to not change the default behavior but I guess we could do this by default.

Reply:
yeah to be honest, I don't see why you would "lookup" for empty templates.

Reply:
One could argue that you just want to get the "parent" template no matter its content in an API. So I'm contradicting myself but I guess I'm ok with both ideas :)

Reply:
I updated to make ignore empty the default behavior.